src/movies.js

```diff
@@ -1,15 +1,105 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
-
+function getAllDirectors(array) {
+    let directorsArr = array.map((movie) => {
+        return movie.director
+    })
+    return directorsArr
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies (array){
+    let dramaMovies = array.filter ((movie) => {
+        return movie.director ==="Steven Spielberg" && movie.genre.includes("Drama")
+    })
+        return dramaMovies.length
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage (movies){
+    if(movies.length===0){
+        return 0
+
+    }else{
+    let total = movies.reduce ((acc, movies) =>{
+        if ( movies.rate == undefined){
+            return acc
+        } else {
+        return acc + movies.rate }
+    },0)
+    let average = total/ movies.length
+    average= average.toFixed(2)
+    average = Number(average)
+    return average
+}
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate (movies) {
+    if (!movies.includes("Drama")){
+        return 0
+    }else { 
+        let TotalDramaRates = movies.reduce ( (acc, movies)=>{
+            if (movies.genre.includes("Drama")){
+                return acc + movie.rate;
```

Comment:
the parameter name is movies, not movie. Sometimes can be tricky using similar names for variables (movies, movie) so it is common to avoid this to name them a little bit different, you could try moviesArray for the array and eachMovie for when working array each object inside the methods.